index.cjs

```diff
@@ -3295,7 +3295,7 @@ class BlockchainServiceBase {
                 to: contractInstance.options.address,
                 data: encodedABI,
                 from: publicKey,
-                gasPrice,
+                gasPrice: isNaN(gasPrice) ? DEFAULT_GAS_PRICE_GWEI.GNOSIS : gasPrice,
```

Comment:
Maybe do `!!gasPrice` instead. To handle more falsy values
```suggestion
                gasPrice: !!gasPrice ? DEFAULT_GAS_PRICE_GWEI.GNOSIS : gasPrice,
```

Reply:
it's okay to get falsy value, then you will retry tx which is okay. Nan is not tho

Reply:
It's not ok, it will last longer and it will happen  again probably and it will fail after multiple retires this approach handles multiple problematic cases